server/src/dtos/search.dto.ts

```diff
@@ -75,34 +75,28 @@ class BaseSearchDto {
   takenAfter?: Date;
 
   @IsString()
-  @IsNotEmpty()
-  @Optional()
-  city?: string;
+  @Optional({ nullable: true })
+  city?: string | null;
 
   @IsString()
-  @IsNotEmpty()
-  @Optional()
-  state?: string;
+  @Optional({ nullable: true })
+  state?: string | null;
 
   @IsString()
-  @IsNotEmpty()
-  @Optional()
-  country?: string;
+  @Optional({ nullable: true })
+  country?: string | null;
 
   @IsString()
-  @IsNotEmpty()
-  @Optional()
+  @Optional({ nullable: true })
   make?: string;
 
   @IsString()
-  @IsNotEmpty()
-  @Optional()
-  model?: string;
+  @Optional({ nullable: true })
+  model?: string | null;
 
   @IsString()
-  @IsNotEmpty()
```

Comment:
I think the request should explicitly be null and not an empty string, so this doesn't need to be removed.

Reply:
I could do that, at least for the web. But the mobile app can't actually send `null` values through the generated client. So either the mobile app doesn't get this functionality or we need to use an empty string in the api to indicate querying for null.